src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -106,7 +112,8 @@ public function __construct(
         ProductColorsRetriever $productColorsRetriever,
         TranslatorInterface $translator,
         HookManager $hookManager = null,
-        Configuration $configuration = null
+        Configuration $configuration = null,
+        EmbeddedAttributesFilter $embeddedAttributesFilter = null
```

Comment:
Done this way to avoid adding a new mandatory constructor argument which would be a BC break

Reply:
Makes sense